pkg/store/cache/matchers_cache.go

```diff
@@ -89,9 +89,9 @@ func NewMatchersCache(opts ...MatcherCacheOption) (*LruMatchersCache, error) {
 
 func (c *LruMatchersCache) GetOrSet(key ConversionLabelMatcher, newItem NewItemFunc) (*labels.Matcher, error) {
```

Comment:
Do we still want `ConversionLabelMatcher` in the method itself?

Reply:
I was thinking the same... maybe better to send the key as string... lemme do that

Reply:
Actually, we need as the key is also used as argument of the `NewItemFunc`. =/

Reply:
@yeya24 PTAL? tried to simplify the code a bit.